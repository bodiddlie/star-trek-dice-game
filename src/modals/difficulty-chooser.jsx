import React from 'react';
import { connect } from 'react-redux';

import { Modal } from './modal';
import { tryChooseDifficulty } from '../sagas/game-start';

export const DifficultyChooser = ({ show, chooseDifficulty }) =>
  <Modal show={show}>
    <button type="button" onClick={() => chooseDifficulty(1)}>
      Easy
    </button>
    <button type="button" onClick={() => chooseDifficulty(2)}>
      Medium
    </button>
    <button type="button" onClick={() => chooseDifficulty(3)}>
      Hard
    </button>
    <button type="button" onClick={() => chooseDifficulty(4)}>
      Kobayashi Maru
    </button>
  </Modal>;

const mapStateToProps = state => ({
  show: state.difficulty === 0,
});

const mapDispatchToProps = dispatch => ({
  chooseDifficulty: difficulty => {
    dispatch(tryChooseDifficulty(difficulty));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(DifficultyChooser);
