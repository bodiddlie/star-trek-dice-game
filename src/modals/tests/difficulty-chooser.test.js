import React from 'react';
import { shallow } from 'enzyme';

import { DifficultyChooser } from '../difficulty-chooser';

describe('<Difficulty Chooser />', () => {
  it('renders a modal for choosing the difficulty', () => {
    const wrapper = shallow(<DifficultyChooser show={true} chooseDifficulty={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
