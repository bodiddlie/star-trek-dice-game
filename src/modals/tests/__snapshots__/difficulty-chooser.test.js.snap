// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Difficulty Chooser /> renders a modal for choosing the difficulty 1`] = `
<Component
  show={true}
>
  <button
    onClick={[Function]}
    type="button"
  >
    Easy
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Medium
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Hard
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Kobayashi Maru
  </button>
</Component>
`;
